Allow reports to have no filters

robincornett opened the pull request:
Fixes #8468

Proposed Changes:
1. Set the default filters property to an empty array.
2. Allow filters to be empty.

To test: add `'filters' => false,` to the [array of attributes for the export report](https://github.com/easydigitaldownloads/easy-digital-downloads/blob/release/3.0/includes/admin/reporting/reports.php#L2526-L2532). You should still have Export available under Reports, and no filters should show.

ashleyfae commented on includes/reports/data/class-report-registry.php:
```diff
@@ -119,7 +119,7 @@ public function add_report( $report_id, $attributes ) {
 
 		try {
-			$this->validate_attributes( $attributes, $report_id );
+			$this->validate_attributes( $attributes, $report_id, array( 'filters' ) );
```
Can you add a comment above that just says filters are excluded because we allow it to be empty?

ashleyfae reviewed (changes requested):
This works to remove filters, but it's revealing that several of our reports are not specifying `dates` as a filter, and as a result, that's disappeared.

We'll need to explicitly set it for all reports that should have it.

robincornett commented:
Updated with the comment, and added `dates` to all reports that should have it and have something other than the default filters set. Removed `filters` parameter from customers and refunds so they would end up with the default (dates and taxes).

ashleyfae commented on includes/admin/reporting/reports.php:
```diff
@@ -2529,6 +2527,7 @@ function edd_register_export_report( $reports ) {
 		    'priority'         => 1000,
 		    'capability'       => 'export_shop_reports',
 		    'display_callback' => 'display_export_report',
+			'filters'          => false,
```
Some wonky spacing here. Looks like it's because the previous arguments use spaces in a place they shouldn't be. Can you adjust that?

ashleyfae reviewed (commented):
Noted one tiny formatting issue, but otherwise this looks good!

ashleyfae commented:
@robincornett Looks like the unit tests need updating. This isn't an issue with your code, but the tests were written to expect `dates` to be the default. We should rework them (and rename them) to expect an empty array instead. Do you feel comfortable digging into that?

ashleyfae commented:
@robincornett Looks like there's one more failure:

```
There was 1 failure:
--
  |  
  | 1) EDD\Reports\Data\Report_Tests::test_get_filters_with_a_valid_non_dates_filter_should_still_include_dates
  | Failed asserting that an array contains 'dates'.
  |  
  | /tmp/xdeTmbBt0tnJhZiPKzP01AKR7ndptVwW/tests/reports/data/tests-report.php:162
```
